### PR TITLE
Add triangular function

### PR DESCRIPTION
**Pull request by gregorybarale**

### CODE REVIEW

File: src/triangular/triangular.spec.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+describe('triangular function', () => {
+  const input = [1, 2, 3, 4, 5];
+
+  test('is a pure function', () => {
+    triangular()(input);
+    expect(input).toMatchSnapshot();
+  });
+
+  test('[1, 2, 3, 4, 5] |> triangular === [[1], [1, 2], [1, 2, 3], [1, 2, 3, 4], [1, 2, 3, 4, 5]]', () => {
+    expect(triangular()(input)).toEqual([
```
Comment (evanliomain) [inline src/triangular/triangular.spec.ts]: Use `.toMatchSnapshot()`

File: src/triangular/triangular.spec.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      [1],
+      [1, 2],
+      [1, 2, 3],
+      [1, 2, 3, 4],
+      [1, 2, 3, 4, 5]
+    ]);
+  });
+});
```
Comment (evanliomain) [inline src/triangular/triangular.spec.ts]: Add a `test('[] |> triangular === [])`

File: src/triangular/triangular.spec.ts (hunk 1 above)
Comment (gregorybarale) [reply src/triangular/triangular.spec.ts]: Done

Comment (gregorybarale) [reply src/triangular/triangular.spec.ts]: Done

**Pull request merged**